### PR TITLE
Android register unregister

### PR DESCRIPTION
**Pull request by kolya182**
PR bring next changes to `unregister` endpoint:
1. `DELETE /push/unregister` => `POST /push/unregister/`.  `DELETE` will be available till mass adoption of `POST` 
2.  Response from `/push/unregister` change from:
```js
{
  status:  boolean,
  ...
}
```
to:
```js
{
  status:  number, // status code
  ...
}
```

### CODE REVIEW

File: src/controllers/DeviceRegistrationController.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    unregister = (req: Request, res: Response) => {
+        const deviceID: string = req.body.deviceID
+        const token: string = req.body.token
+        const type: string = req.body.type
+
+        try {
+            if (deviceID) {
+                Device.findOneAndRemove({deviceID}).then((removedDevice) => {
+                    removedDevice ? this.sendOnUnregisterSuccess(res, removedDevice) : this.sendOnUnregisterFail(404, res, "Can not find device", removedDevice)
```
Comment (maratsubkhankulov) [inline src/controllers/DeviceRegistrationController.ts]: you can deduplicate this and the else block by aliasing either deviceID or token as a new "id" variable.

File: src/controllers/DeviceRegistrationController.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            wallets,
+            token,
+            preferences,
+            type
+        }
+
+        try {
+            if (deviceID && token) {
+                Device.findOneAndUpdate({deviceID}, updatesParams, updateOptions).then(registeredDevice => {
+                    this.sendOnRegisterSuccess(res, registeredDevice)
```
Comment (maratsubkhankulov) [inline src/controllers/DeviceRegistrationController.ts]: the if and else blocks are duplicate code, you can minimize it.

File: src/controllers/PusherController.ts [hunk 3]
```diff
@@ -6,41 +6,49 @@ import { Error } from "mongoose";
...
+            wallets,
+            token,
+            preferences,
+            type
+        }
+
+        try {
+            if (deviceID && token) {
+                Device.findOneAndUpdate({deviceID}, updatesParams, updateOptions).then(registeredDevice => {
+                    this.sendOnRegister(res, registeredDevice)
```
Comment (maratsubkhankulov) [inline src/controllers/PusherController.ts]: ditto - see previous comment

File: src/controllers/DeviceRegistrationController.ts (hunk 1 above)
Comment (kolya182) [reply src/controllers/DeviceRegistrationController.ts]: Done

File: src/controllers/DeviceRegistrationController.ts (hunk 2 above)
Comment (kolya182) [reply src/controllers/DeviceRegistrationController.ts]: Done

File: src/controllers/PusherController.ts (hunk 3 above)
Comment (kolya182) [reply src/controllers/PusherController.ts]: Done

**Comment by maratsubkhankulov** [comment]
Agreed on keeping both DELETE/POST - maintaining backwards compatibility is a good thing.

**Pull request merged**